src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -0,0 +1,558 @@
+
+    cofactorBn = (*env)->CallStaticObjectMethod(env, g_bigNumClass, g_valueOfMethod, (int64_t)cofactorInt);
+
+    // Extract seed (optional)
+    seedArray = (*env)->CallObjectMethod(env, group, g_EllipticCurveGetSeed);
+    if (seedArray)
+    {
+        seedBn = (*env)->NewObject(env, g_bigNumClass, g_bigNumCtor, seedArray);
+
+        *seed = ToGRef(env, seedBn);
+        *cbSeed = CryptoNative_GetBigNumBytes(seedBn);
+
+        /*
+            To implement SEC 1 standard and align to Windows, we also want to extract the nid
+            to the algorithm (e.g. NID_sha256) that was used to generate seed but this
+            metadata does not appear to exist in openssl (see openssl's ec_curve.c) so we may
+            eventually want to add that metadata, but that could be done on the managed side.
+        */
+    }
+    else
+    {
+        *seed = NULL;
+        *cbSeed = 0;
+    }
+
+    // Success; assign variables
+    *gx = ToGRef(env, xBn); *cbGx = CryptoNative_GetBigNumBytes(xBn);
+    *gy = ToGRef(env, yBn); *cbGy = CryptoNative_GetBigNumBytes(yBn);
+    *p = ToGRef(env, pBn); *cbP = CryptoNative_GetBigNumBytes(pBn);
+    *a = ToGRef(env, aBn); *cbA = CryptoNative_GetBigNumBytes(aBn);
+    *b = ToGRef(env, bBn); *cbB = CryptoNative_GetBigNumBytes(bBn);
+    *order = ToGRef(env, orderBn); *cbOrder = CryptoNative_GetBigNumBytes(orderBn);
+    *cofactor = ToGRef(env, cofactorBn); *cbCofactor = CryptoNative_GetBigNumBytes(cofactorBn);
+
+    rc = SUCCESS;
+    
+    ReleaseLRef(env, xBn);
+    ReleaseLRef(env, yBn);
+    ReleaseLRef(env, pBn);
+    ReleaseLRef(env, aBn);
+    ReleaseLRef(env, bBn);
+    ReleaseLRef(env, orderBn);
+    ReleaseLRef(env, cofactorBn);
+    ReleaseLRef(env, seedBn);
+
+    goto exit;
+
+error:
+    // Clear out variables from CryptoNative_GetECKeyParameters
+    *cbQx = *cbQy = 0;
+    ReleaseGRef(env, *qx);
+    ReleaseGRef(env, *qy);
+    *qx = *qy = NULL;
+    if (d)
+    {
+        ReleaseGRef(env, *d);
+        *d = NULL;
+    }
+    if (cbD) *cbD = 0;
+
+    // Clear our out variables
+    *curveType = Unspecified;
+    *cbP = *cbA = *cbB = *cbGx = *cbGy = *cbOrder = *cbCofactor = *cbSeed = 0;
+    ReleaseGRef(env, *p);
+    ReleaseGRef(env, *a);
+    ReleaseGRef(env, *b);
+    ReleaseGRef(env, *gx);
+    ReleaseGRef(env, *gy);
+    ReleaseGRef(env, *order);
+    ReleaseGRef(env, *cofactor);
+    ReleaseGRef(env, *seed);
+    *p = *a = *b = *gx = *gy = *order = *cofactor = *seed = NULL;
+
+    ReleaseLRef(env, xBn);
+    ReleaseLRef(env, yBn);
+    ReleaseLRef(env, pBn);
+    ReleaseLRef(env, aBn);
+    ReleaseLRef(env, bBn);
+    ReleaseLRef(env, orderBn);
+    ReleaseLRef(env, cofactorBn);
+    ReleaseLRef(env, seedBn);
+
+exit:
+    return rc;
+}
+
+static jobject CryptoNative_CreateKeyPairFromCurveParameters(jobject curveParameters, uint8_t* qx, int32_t qxLength, uint8_t* qy, int32_t qyLength, uint8_t* d, int32_t dLength)
```

Comment:
At one point we were trying to be good at running `clang-format -i <file>` on the shim files.  I'm pretty sure it will rewrite this into a style I don't personally like, but it does provide some uniformity.